stream/rlog/rlog.go

```diff
@@ -0,0 +1,260 @@
+package rlog
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"unsafe"
+
+	"github.com/sei-protocol/sei-db/proto"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/wal"
+)
+
+type Manager struct {
+	rlog         *wal.Log
+	writeChannel chan *LogEntry
+	quitSignal   chan error
+	config       Config
+}
+
+type Config struct {
+	Fsync            bool
+	ZeroCopy         bool
+	WriteChannelSize int
+}
+
+type LogEntry struct {
+	Index uint64
+	Data  proto.ReplayLogEntry
+}
+
+func NewManager(dir string, config Config) (*Manager, error) {
```

Comment:
Can we add some more comments here to `Manager` and `NewManager`. Just for others to have context on what each do and motivation

Reply:
Added